source/common/router/router.cc

```diff
@@ -381,6 +404,24 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
   ENVOY_STREAM_LOG(debug, "cluster '{}' match for URL '{}'", *callbacks_,
                    route_entry_->clusterName(), headers.Path()->value().getStringView());
 
+  if (config_.strict_check_headers_ != nullptr && !config_.strict_check_headers_->empty()) {
```

Comment:
I think you can omit the empty check since the for loop will just noop in that case